packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -433,18 +442,31 @@ class _RunAndStayResident {
     }
 
     if (benchmark) {
-      _writeBenchmark(stopwatch);
-      new Future<Null>.delayed(new Duration(seconds: 2)).then((_) {
-        _handleExit();
-      });
+      await new Future<Null>.delayed(new Duration(seconds: 4));
+
+      // Touch the file.
+      File mainFile = new File(mainPath);
+      mainFile.writeAsBytesSync(mainFile.readAsBytesSync());
+
+      Stopwatch restartTime = new Stopwatch()..start();
+      bool restarted = await _handleRefresh(package, result, mainPath);
+      restartTime.stop();
+      _writeBenchmark(startTime, restarted ? restartTime : null);
+      await new Future<Null>.delayed(new Duration(seconds: 2));
+      _handleExit();
     }
 
     return _exitCompleter.future.then((int exitCode) async {
-      if (observatory != null && !observatory.isClosed && _isolateId != null) {
-        observatory.flutterExit(_isolateId);
-
-        // WebSockets do not have a flush() method.
-        await new Future<Null>.delayed(new Duration(milliseconds: 100));
+      try {
+        if (observatory != null && !observatory.isClosed) {
+          if (observatory.isolates.isNotEmpty) {
+            observatory.flutterExit(observatory.firstIsolateId);
+            // The Dart WebSockets API does not have a flush() method.
```

Comment:
Is there a bug we can track? Maybe provide a link to the relevant SDK github issue?